### PR TITLE
Add Red Hat to vendors

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

File: data/ecosystem/vendors.yaml [hunk 1]
```diff
@@ -251,6 +251,13 @@
   contact: ''
   oss: false
   commercial: true
+- name: Red Hat
+  distribution: true
+  nativeOTLP: true
+  url: 'https://docs.openshift.com/container-platform/4.14/otel/otel-release-notes.html'
+  contact: ''
```
Comment (svrnm) [inline data/ecosystem/vendors.yaml:258]: please add a point of contact, can be your github handle or an email address, we may use this eventually if we need to reach out

Comment (pavolloffay) [reply data/ecosystem/vendors.yaml:258]: Is it for reaching out to maintainers or sales?

Comment (svrnm) [reply data/ecosystem/vendors.yaml:258]: Maintainers.

**Pull request merged**